service/lib/dinstaller/config_reader.rb

```diff
@@ -105,37 +112,25 @@ def copy_file(location, target)
       res
     end
 
-    def cmdline_reader
-      return @cmdline_reader if @cmdline_reader
-
-      @cmdline_reader = CmdlineArgs.new(workdir: workdir)
-      @cmdline_reader.read
-      @cmdline_reader
+    # @return [CmdlineArgs]
+    def cmdline_args
+      @cmdline_args ||= CmdlineArgs.read_from(File.join(workdir, "/proc/cmdline"))
     end
 
     # return [Config]
     def cmdline_config
-      return if cmdline_reader.args.empty?
-
-      Config.new(cmdline_reader.args)
+      Config.new(cmdline_args.data)
     end
 
     # return [Config]
     def remote_config
-      return unless cmdline_reader.config_url
+      return unless cmdline_args.config_url
 
       file_path = File.join(Yast::Directory.tmpdir, REMOTE_BOOT_CONFIG)
       logger.info "Copying boot config to #{file_path}"
 
-      copy_file(cmdline_reader.config_url, file_path)
-      config_from_file(file_path)
-    end
-
-    # loads correct yaml file
-    def config_from_file(path = nil)
-      raise "Missing config file at #{path}" unless File.exist?(path)
-
-      Config.new(YAML.safe_load(File.read(path)))
+      copy_file(cmdline_args.config_url, file_path)
+      self.class.from_file(file_path)
```

Comment:
Sorry, my previous comment was confusing (and wrong): having an instance method was actually better. I meant a `Config.from_file`, but as we have a `ConfigReader`, it is not that useful. Your previous version was better :-)